pkg/flow/logging/logger.go

```diff
@@ -85,11 +92,25 @@ func (l *Logger) Update(o Options) error {
 	}
 	l.writer.Set(newWriter)
 
+	// Print out the buffered logs since we determined the log format already
+	for _, bufferedLogChunk := range l.buffer {
+		if err := slogadapter.GoKit(l.handler).Log(bufferedLogChunk...); err != nil {
+			return err
+		}
+	}
```

Comment:
the buffer should be emptied here, else every call to Update() will print the content of the buffer
`l.buffer = nil`